hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/cache/HierarchicalCacheLevel.java

```diff
@@ -36,11 +42,14 @@
                            String path,
                            int depth,
                            CacheListeners eventConsumer,
-                           Optional<BiFunction<Integer, String, HierarchicalCacheLevel>> nextLevelFactory) {
+                           Optional<BiFunction<Integer, String, HierarchicalCacheLevel>> nextLevelFactory,
```

Comment:
Maybe it would be better to create some value object instead of using `Optional<BiFunction<Integer, String, HierarchicalCacheLevel>>`